components/common/markdown/markdown-editor.tsx

```diff
@@ -0,0 +1,50 @@
+'use client';
+
+import React, { useState } from 'react';
+import MDEditor from '@uiw/react-md-editor';
+import Markdown from './markdown';
+import { Label } from '@/components/ui/label';
+import Button from '@/components/common/buttons/button';
+import { cn } from '@/libs/utils';
+
+type MarkdownEditorProps = React.ComponentProps<typeof MDEditor> & {
+  headerStyle?: string;
+  label: string;
+};
+
+export default function MarkdownEditor({
+  label,
+  headerStyle,
+  value,
+  onChange,
+  ...props
+}: MarkdownEditorProps) {
+  const [isPreview, setIsPreview] = useState(false);
+
+  const toggle = () => setIsPreview(!isPreview);
```

Comment:
p5: 사소한데 저는 이런 코드는 updater 함수를 쓰려고 하고 있어요..!

```suggestion
  const toggle = () => setIsPreview((prev) => !prev);
```